File: El Regalo de Leela/El Regalo de Leela.py
#define Hd
def abs(a:int) -> int:
    return a if a>=0 else -a

n = int(input())
mp : int = 0
ar = list(map(int,input().split()))
for i in range(1,n):
    if ar[i] > ar[mp]:
        mp = i

aux : int = ar[mp]//2
posa : int = -1
ans : int
for i in range(n):
    if i!=mp:
        if posa==-1:
            ans = ar[i]
            posa = 1
        else:
            a,b = abs(ans-aux),abs(ar[i]-aux)
            if a>b:
                ans = ar[i]
            elif a==b and ans<ar[i]:
                ans = ar[i]

print(ar[mp],ans)

File: El Regalo de Leela/ El Regalo de Leela.cpp
#include<stdio.h>
#define Hd main(void)

int ABS(int a){
    return (a>0? a:a*-1);
}
Hd { 
    int n;
    scanf("%d",&n);
    int mp=0;
    int ar[n];
    scanf("%d",&ar[0]);
    for(int i=1; i<n; i++) {
        scanf("%d",&ar[i]);
        if(ar[i]>ar[mp]) {
            mp = i;
        }
    }
    int aux=ar[mp]/2,posa=-1,ans;
    for(int i=0; i<n; i++) {
        if(i!=mp){
            if(posa==-1){
                ans=ar[i];
                posa=1;
            }
            else{
                int a=ABS(ans-aux),b=ABS(ar[i]-aux);
                if(a>b){
                    ans=ar[i];
                }
                else if(a==b&&ans<ar[i]){
                    ans=ar[i];
                }
            }
        }
    }
    printf("%d %d",ar[mp],ans);
}
